src/main/resources/XWiki/SuggestDisplayEditItems.xml

```diff
@@ -507,8 +513,26 @@ initializeAttachmentPicker : function(itemEditURL,attPickerURL,XWModalContent) {
                      success: function(){
                         // Reload the page after upload
                         $(".xnotification-inprogress").remove();
-                        $.initializeAttachmentPicker(itemEditURL,attPickerURL,XWModalContent);  
-                     }
+                        if (!isNewItem) {
+                          $.initializeAttachmentPicker(itemEditURL,attPickerURL,XWModalContent,isNewItem);
+                        }
+                       else
+                        {
+                           // Save the item, close the attachmeents selector popup and show the item edit form
+                           var itemSaveURL = itemEditURL.replace('/edit/','/save/') + "&amp;form_token=" + "$services.csrf.getToken()";
```

Comment:
Bad idea to use `$services.csrf.getToken()` there. The javascript file could be cached by the browser with a outdated token.

Please use: http://platform.xwiki.org/xwiki/bin/view/DevGuide/JavaScriptAPI#HGetsomeinformationaboutthecurrentdocument

For version < 6.3, use (but it is deprecated...):
`$('meta[name="form_token"]').val()` (jquery)
or
`$$('meta[name="form_token"]')[0].content` (prototype)

Ideally, you should test if `require['xwiki-meta']` is available or not and use it when you can.